audio-worklet/src/noisegate-audio-worklet.js

```diff
@@ -0,0 +1,194 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @class NoiseGate
+ * @extends AudioWorkletProcessor
+ * @description A noise gate allows audio signals to pass only when the
+ * registered volume is above a specified threshold.
+ */
+registerProcessor('noisegate-audio-worklet',
+                  class NoiseGateAudioWorklet extends AudioWorkletProcessor {
+  
+  static get parameterDescriptors() {
+    return [{
+      name: 'attack',
+      defaultValue: 0,
+      minValue: 0,
+      maxValue: 10
+    }, {
+      name: 'release',
+      defaultValue: 0,
+      minValue: 0,
+      maxValue: 10
+    }, {
+      name: 'threshold',
+      defaultValue: -40,
+      minValue: -100,
+      maxValue: 0
+    }, {
+      name: 'timeConstant',
+      defaultValue: 0.0025,
+      minValue: 0,
+      maxValue: 1
+    }];
+  }
+
+  constructor() {
+    super();
+
+    // The previous envelope level, a float representing signal amplitude.
+    this.previousLevel_ = 0;
+
+    // The last weight (between 0 and 1) assigned, where 1 means the gate
+    // is open and 0 means it is closed and the sample in the output buffer is
+    // muted. When attacking, the weight will linearly decrease from 1 to 0, and
+    // when releasing the weight linearly increase from 0 to 1.
+    this.previousWeight_ = 1.0;
+    this.envelope_ = new Float32Array(128);
+    this.weights_ = new Float32Array(128);
+    this.sampleRate = 44100;
+  }
+
+  /**
+   * Control the dynamic range of input based on specified threshold.
+   * @param {AudioBuffer} input Input audio data
+   * @param {AudioBuffer} output Output audio data
+   * @param {Object} parameters
+   * @param {Number} parameters.attack Seconds for gate to fully close.
+   * @param {Number} parameters.release Seconds for gate to fully open.
+   * @param {Number} parameters.timeConstant Seconds for envelope follower's
+   *                                         smoothing filter delay.
+   * @param {Number} parameters.threshold Decibel level beneath which sound is
+   *                                      muted.
+   */
+  process(input, output, parameters) {
+    // Alpha controls a tradeoff between the smoothness of the
+    // envelope and its delay, with a higher value giving more smoothness at
+    // the expense of delay and vice versa.
+    this.alpha_ = this.getAlphaFromTimeConstant_(
+        parameters.timeConstant[0], 44100);
+
+    this.attack = parameters.attack[0];
```

Comment:
I should say that the a-rate audio params are used as k-rate audio params since the current build does not support k-rate audio params?